### PR TITLE
Listing table

### PR DESCRIPTION
**Pull request by adriencyberspace**
# Pull Request Template

## Description

This PR refactors uses of a global `<table>` element and leans more on DataTable component. It also pulls css out of global stylesheets and into module scss files. 

## How Can This Be Tested/Reviewed?

Visit any /service/SRVID page and check that the tables are uniform (except the table in Accordion which also has its own css pulled into Accordion.module.scss)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have added QA notes to the issue with applicable URLs
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [x] I have reviewed the changes in a mobile view
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have assigned reviewers
- [x] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

File: app/components/listing/ContactInfoTableRows.tsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
+import React from "react";
+import { Service } from "../../models";
+
+export const ContactInfoTableRows = ({ service }: { service: Service }) => {
+  const website = service.url || service.resource?.website;
+  const email = service.email || service.resource?.email;
+  const phones = service.resource?.phones;
+
+  if (!website && !phones && !email) {
```
Comment (rosschapman) [inline app/components/listing/ContactInfoTableRows.tsx:9]: Are you checking if `phones` is null here? The `Organization` type suggests `phones` will always be an array. If in experience this value can return null from the api can you update the type please? 

🏓 This array bool check is an alternative I like if you think this list will ever expand: 
```ts
[website, phones, email].every(Boolean)
```

File: app/components/listing/ContactInfoTableRows.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+  return (
+    <>
+      {website && (
+        <tr>
+          <th>Website</th>
+          <td>
+            <a target="_blank" rel="noopener noreferrer" href={website}>
+              {website}
+            </a>
```
Comment (rosschapman) [inline app/components/listing/ContactInfoTableRows.tsx:28]: Suggestion: A `<TargetBlankLink />` might be a useful component to reduce some keystrokes.

File: app/components/listing/ContactInfoTableRows.tsx [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+      {phones?.length && (
+        <tr>
+          <th>Phone</th>
+          <td>
+            <ul>
+              {phones.map((phone) => (
+                <li key={phone.number}>
+                  <a href={`tel:${phone.number}`}>{phone.number}</a>{" "}
+                  {phone.service_type && `(${phone.service_type})`}
```
Comment (rosschapman) [inline app/components/listing/ContactInfoTableRows.tsx:50]: 🤔 If this was written `{phone?.service_type}` without the operand and RHS of the expression would it accomplish the same thing?

**Review by rosschapman — APPROVED** [review]
@adriencyberspace a very reasonable refactor. I'm curious, is the `DataTable` component meant for reuse? It seems configured and styled fairly specifically for the Service Listing page. Can this also replace the `DataTable` in `app/pages/Pdf/ServicePdfPage.tsx`? 

These are curiosities we can address later, I know a lot of this code is pre-existing. I also find the name `DataTable` a little confusing because I've seen that refer to more sophisticated table grids, not a more simplistic contact card. That makes me wonder if we'd eventually want to use the `<address />` tag for better semantics for Service Listing card. Theoretically that improves accessibility as well?

**Comment by adriencyberspace** [comment]
I agree with the above notes! Changed the DataTable name to ListingInfoTable. The other notes are on pre-existing code and I do agree but I think additional code cleanup should be a different ticket.

**Pull request merged**